### PR TITLE
Organizations permissions

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?239222

Limits editing structure of locations to editors of mobile works and editing levels to app editors

buddy @czue

### CODE REVIEW

**Comment by czue** [comment]
@sheelio are we documenting these restrictions anywhere? seems complicated/non-intuitive enough that it would be good to do.

File: corehq/apps/locations/views.py [hunk 1]
```diff
@@ -122,6 +123,7 @@ class LocationsListView(BaseLocationView):
 
     @use_jquery_ui
     @location_safe
+    @method_decorator(require_can_edit_commcare_users)
```
Comment (esoergel) [inline corehq/apps/locations/views.py]: @sheelio are you referring to the "Users" tab of an "Edit Location" page?  If so, I believe this check needs to go at the top of this method: https://github.com/dimagi/commcare-hq/blob/f806fa5e06fd86dc999ecdefec2cf3621da555f0/corehq/apps/locations/views.py#L602-L615

**Comment by sheelio** [comment]
@czue 
WIll document the restrictions on the web user page.

File: corehq/apps/locations/views.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/locations/views.py]: @sheelio and I offlined, and it sounds like this should go in `locations_access_required`, as no locations views should be accessible to those users.

**Pull request merged**